sklearn/cluster/_kmeans.py

```diff
@@ -395,14 +396,24 @@ def _kmeans_single_elkan(X, sample_weight, centers_init, max_iter=300,
 
         centers, centers_new = centers_new, centers
 
-        center_shift_tot = (center_shift**2).sum()
-        if center_shift_tot <= tol:
-            if verbose:
-                print(f"Converged at iteration {i}: center shift "
-                      f"{center_shift_tot} within tolerance {tol}.")
-            break
+        if tol == 0:
```

Comment:
I think this should also be used for other values of tol.
Consider the case where a really small value of tol is set, say, 1e-100.
In such cases, it should nevertheless stop if labels do not change, even if rounding causes the computed tol to be larger than 1e-100

Reply:
I don't see why someone would put a tol of 1e-100 but fine :)
I changed it